src/components/frontpage/Countdown.astro

```diff
@@ -0,0 +1,45 @@
+---
+const { when } = Astro.props;
+---
+
+<script define:vars={{ when }}>
+  const countdownElement = document.currentScript.parentElement;
+  let countdown = { days: 0, hours: 0, minutes: 0, seconds: 0 };
+  const targetDate = new Date(when).getTime();
+
+  function updateCountdown() {
+    const now = new Date().getTime();
+    const distance = Math.max(0, targetDate - now);
+
+    countdown.days = Math.floor(distance / (1000 * 60 * 60 * 24));
+    countdown.hours = Math.floor(
+      (distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60),
+    );
+    countdown.minutes = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60));
+    countdown.seconds = Math.floor((distance % (1000 * 60)) / 1000);
+
+    countdownElement.querySelectorAll("[data-countdown]").forEach((el) => {
+      el.textContent = countdown[el.dataset.countdown];
+    });
```

Comment:
We do need some basic timer ended handling (as we probably don't want to try to sync tg.no releases with calendar/show times). Perhaps adding a "ended" slot, and switch to that html (and disconnect any timers) when done?